### ISSUE TITLE
Form Api - `addValue` not working as expected

### ISSUE THREAD

**Issue by AverageZ** [react-tools/react-form#58]
Version: `"react-form": "^1.2.6"`

### Overview

I've been referencing the annotated example as I build a form that contains a nested form. The nested form is a list of items called `phonenumbers` (like 'friends' in the example). Outside of the nested form is an element that uses the `addValue` method from the library. The goal is to use the add button to add a new item to the list which will then get rendered etc.

### The issue
When I click the `addValue` button, it resets the value of `phonenumbers` instead of pushing or appending the `newBasePhone` object

-----

In the following example:

* `phonenumbers` is a list of objects with keys like: "value", "label", "type"
* `newBasePhone` is an object like the ones in `phonenumbers` and has a "label" of "new"

The main form referencing the nested form and the Add button
```jsx
<NestedForm
  field="phonenumbers"
>
  {SmartPhoneInput}
</NestedForm>

<ActionsWrapper>
  <Button
    type="primary"
    onClick={() => addValue('phonenumbers', newBasePhone)}
  >
    Add Number
  </Button>
</ActionsWrapper>
```

The `SmartPhoneInput` nested form spitting out just the label for now.
```jsx
const SmartPhoneInput = (
  <Form>
    {({ values }) => (
      <Wrapper>
        {
          map(values, (number, index) => (
            <p>{number.label}</p>
          ))
        }
      </Wrapper>
    )}
  </Form>
);
```

![apr-20-2017 10-53-27](https://cloud.githubusercontent.com/assets/9088088/25237072/a44f1a72-25b7-11e7-9765-790b6f1fd636.gif)

**Comment by tannerlinsley** [comment]
We'll probably need to see more code to know what is happening.  If you're not comfortable posting that much code here, you could also post to the [slack org](https://react-chat-signup.herokuapp.com/)'s #react-form channel

**Comment by casperOne** [comment]
@tannerlinsley Here's a complete example on codesandbox.io:

https://codesandbox.io/s/ko3no7062r

If you click on "Add Child" you'll see that the text fields added do not display the `childId` field that is in the object passed to the call to `addValue`.

However, if you place a breakpoint somewhere in `MyForm` you will see that after "Add Child" is clicked, the value exists in the array for one iteration of the render of `MyForm` and then the element in the array is set to `undefined`.

![temp](https://user-images.githubusercontent.com/561862/32151349-986b7714-bcf2-11e7-992f-92b0c7a5e920.gif)

**Comment by joepuzzo** [comment]
We have moved to v2 of this lib. I would upgrade and check out the new docs

**Comment by tannerlinsley** [comment]
This is now working as expected. Closing for now.

**Comment by casperOne** [comment]
@tannerlinsley Thanks for taking a look.

Any window on when it may be released?  Is it fixed in just the latest version, or will it be addressed in 1.3 as well?

(sorry, I don't see any changes related to this on npmjs.org, no commits, no changes to the codesandbox.io example)

**Comment by tannerlinsley** [comment]
In the next minor release, most likely. Stay tuned.
